docs/user-guide/docker/building-in-docker.rst

```diff
@@ -0,0 +1,244 @@
+For commands which require interactive input, like ``rasa shell`` and ``rasa interactive``,
+you need to pass the ``-it`` flags.
+
+
+Customizing your Model
+**********************
+
+Choosing a Tag
+##############
+
+To keep images as small as possible, we publish different tags of the ``rasa/rasa`` image
+with different dependencies installed. See :ref:`choosing-a-pipeline` for more information
+about depedencies.
+
+All tags start with a version -- the ``latest`` tag corresponds to the current master build.
+The tags are:
+
+- ``{version}``
+- ``{version}-spacy-en``
+- ``{version}-spacy-de``
+- ``{version}-mitie-en``
+- ``{version}-full``
+
+The plain ``{version}`` tag includes all the dependencies you need to run the ``supervised_embeddings`` pipeline.
+If you are using components with pre-trained word vectors, you need to choose the corresponding tag.
+Alternatively, you can use the ``-full`` tag, which includes all pipeline dependencies.
+
+.. note::
+
+   You can see a list of all the versions and tags of the Rasa Docker image
+   `here <https://hub.docker.com/r/rasa/rasa/>`_.
+
+
+.. _model_training_docker:
+
+Training a Custom Rasa Model with Docker
+########################################
+
+Edit the ``config.yml`` file to use the pipeline you want, and place
+your NLU and Core data into the ``data/`` directory.
+Now you can train your Rasa model by running:
+
+.. code-block:: bash
+
+  docker run \
+    -v $(pwd):/app \
+    rasa/rasa:latest-full \
+    train \
+      --domain domain.yml \
+      --data data \
+      --out models
+
+Here's what's happening in that command:
+
+  - ``-v $(pwd):/app``: Mounts your project directory into the Docker
+    container so that Rasa can train a model on your training data
+  - ``rasa/rasa:latest-full``: Use the Rasa image with the tag ``latest-full``
+  - ``train``: Execute the ``rasa train`` command within the container. For more
+    information see :ref:`command-line-interface`.
+
+In this case, we've also passed values for the location of the domain file, training
+data, and the models output directory to show how these can be customized.
+You can also leave these out since we are passing the default values.
+
+.. note::
+
+    If you are using a custom NLU component or policy, you have to add the module file to your
+    Docker container. You can do this by either mounting the file or by including it in your
+    own custom image (e.g. if the custom component or policy has extra dependencies). Make sure
+    that your module is in the Python module search path by setting the
+    environment variable ``PYTHONPATH=$PYTHONPATH:<directory of your module>``.
+
+Adding Custom Actions
+*********************
+
+To create more sophisticated assistants, you will want to use :ref:`custom-actions`.
+Continuing the example from above, you might want to add an action which tells
+the user a joke to cheer them up.
+
+Running a Custom Action in Docker-Compose
+#########################################
```

Comment:
this probably shouldn't say docker-compose right?  at this point we're just building with docker?

Reply:
i can see a reference to docker-compose below as well.